ci/cudf_pandas_scripts/fetch_pandas_versions.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import requests
+from packaging.version import Version
+from packaging.specifiers import SpecifierSet
+import argparse
+
+def get_pandas_versions(pandas_range):
+    url = "https://pypi.org/pypi/pandas/json"
+    response = requests.get(url)
+    data = response.json()
+    versions = data['releases'].keys()
+
+    specifier = SpecifierSet(pandas_range.lstrip("pandas"))
+
+    minor_versions = list(set([version[:3] for version in versions if Version(version) in specifier]))
+
+    return minor_versions
```

Comment:
I didn't realize this at first -- this is doing string parsing of versions and taking only the first three characters, which is fragile against things like `2.10`. Let's avoid that. We should also sort the versions least to greatest. Here's my proposal:
```suggestion
def get_pandas_versions(pandas_range):
    url = "https://pypi.org/pypi/pandas/json"
    response = requests.get(url)
    data = response.json()
    versions = [Version(v) for v in data['releases']]
    specifier = SpecifierSet(pandas_range.lstrip("pandas"))
    matching_versions = [v for v in versions if v in specifier]
    matching_minors = sorted(set(".".join((str(v.major), str(v.minor))) for v in matching_versions), key=Version)
    return matching_minors
```